snippets/base/admin/fields.py

```diff
@@ -26,3 +29,46 @@ def to_jexl(self, value):
         if value:
             return '{} == {}'.format(self.attr_name, value)
         return None
+
+
+class JEXLRangeField(MultiValueField):
+    def __init__(self, attr_name, *args, **kwargs):
+        self.attr_name = attr_name
+        self.jexl = {
+            'minimum': '{minimum} <= {attr_name}',
+            'maximum': '{attr_name} < {maximum}'
+        }
+        self.jexl = kwargs.pop('jexl', self.jexl)
+        choices = kwargs.pop('choices')
+        fields = (
+            ChoiceField(choices=choices),
+            ChoiceField(choices=choices),
+        )
+        kwargs['widget'] = RangeWidget(choices=choices)
```

Comment:
Just checking if we should ensure the JEXL coming in from `kwargs` has all the necessary things.

Also, should we ensure `choices` exists in `kwargs`?

(It's pretty obvious that all information is present in this PR, I'm just wondering about 6 months down the road when this gets even more complex.)

Reply:
Good point. I'll make choices a separate argument.